engine/src/main/scala/cromwell/engine/workflow/ShadowWorkflowManagerActor.scala

```diff
@@ -0,0 +1,255 @@
+package cromwell.engine.workflow
+
+import akka.actor.FSM.{CurrentState, SubscribeTransitionCallBack, Transition}
+import akka.actor._
+import akka.event.Logging
+import com.typesafe.config.{Config, ConfigFactory}
+import cromwell.core.WorkflowId
+import cromwell.engine._
+import cromwell.engine.backend._
+import cromwell.engine.db.DataAccess._
+import cromwell.engine.workflow.ShadowWorkflowActor._
+import cromwell.engine.workflow.ShadowWorkflowManagerActor.AbortWorkflowCommand
+import cromwell.engine.workflow.ShadowWorkflowManagerActor._
+import cromwell.webservice.CromwellApiHandler._
+import lenthall.config.ScalaConfig.EnhancedScalaConfig
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future, Promise}
+import scala.language.postfixOps
+import scala.util.{Failure, Success}
+
+object ShadowWorkflowManagerActor {
+
+  case class WorkflowIdToActorRef(workflowId: WorkflowId, workflowActor: ActorRef)
+
+  sealed trait ShadowWorkflowManagerActorMessage
+  /**
+    * Commands
+    */
+  sealed trait ShadowWorkflowManagerActorCommand extends ShadowWorkflowManagerActorMessage
+
+  case class SubmitWorkflowCommand(source: WorkflowSourceFiles) extends ShadowWorkflowManagerActorCommand
+  // TODO: This SubscribeToWorkflowCommand is a candidate for deletion (see the handler if you don't believe me!)
```

Comment:
I wouldn't worry too much about it, it's just one if those patterns I'd like people to think carefully about prior to adding to the mayhem. Otoh if you see an easy way to avoid it go for it